interfaces/vault_kv/v0/README.md

```diff
@@ -0,0 +1,62 @@
+# `vault-kv`
+
+## Usage
+
+Some charms require a secure key value store. This relation interface describes the expected behavior of any charm claiming to interact with Vault Key Value stores.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Requirer -- mount_suffix, egress_subnet --> Provider
+    Provider -- vault_url, mount, credentials --> Requirer
+```
+
+## Behavior
+
+Both the Requirer and the Provider need to adhere to criteria to be considered compatible with the interface.
+
+### Provider
+
+Provider expectations
+
+- Is expected to provide the vault url
+- Is expected to provide a key value mount, the mount name shall respect the following pattern: `charm-<requirer app>-<user provided suffix>`
+- Is expected to create an approle restricted to the requiring unit's egress subnet.
```

Comment:
The implication of this statement is that the AppRole must be updated with a remote units new /32 CIDR when a pod gets recreated and its IP changes - how do we coordinate this with the requiring side unit to ensure that the consuming service is not started before the AppRole has been updated?

Reply:
I don't see much options, either we accept the requiring unit having a wrong set of credentials while the providing app is creating the new ones, or we should send back the unit's egress subnet (either in the secret or in databag) so that the requiring unit can make sure the credentials is valid.

Reply:
@javacruft any input ?

Reply:
I think as long as the consuming app understands there will be a period of time during which the old credentials will not be usable that's OK - we should make sure that's documented in the interface.

Charms can then deal with themselves.